catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -0,0 +1,568 @@
+import org.codice.ddf.catalog.async.data.api.internal.ProcessRequest;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessResource;
+import org.codice.ddf.catalog.async.data.api.internal.ProcessUpdateItem;
+import org.codice.ddf.catalog.async.data.impl.ProcessCreateItemImpl;
+import org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl;
+import org.codice.ddf.catalog.async.data.impl.ProcessUpdateItemImpl;
+import org.codice.ddf.catalog.async.plugin.api.internal.PostProcessPlugin;
+import org.codice.ddf.platform.util.TemporaryFileBackedOutputStream;
+import org.codice.imaging.nitf.core.common.NitfFormatException;
+import org.codice.imaging.nitf.core.image.ImageSegment;
+import org.codice.imaging.nitf.fluent.impl.NitfParserInputFlowImpl;
+import org.codice.imaging.nitf.render.NitfRenderer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This pre-storage plugin creates and stores the NITF thumbnail and NITF overview images. The
+ * thumbnail is stored with the Metacard while the overview is stored in the content store.
+ */
+public class NitfPostProcessPlugin implements PostProcessPlugin {
+
+  @VisibleForTesting static final double DEFAULT_MAX_SIDE_LENGTH = 1024.0;
+
+  private static final String IMAGE_NITF = "image/nitf";
+
+  static final MimeType NITF_MIME_TYPE;
+
+  static {
+    IIORegistry.getDefaultInstance().registerServiceProvider(new J2KImageReaderSpi());
+
+    try {
+      NITF_MIME_TYPE = new MimeType(IMAGE_NITF);
+    } catch (MimeTypeParseException e) {
+      throw new ExceptionInInitializerError(
+          String.format("Unable to create MimeType from '%s': %s", IMAGE_NITF, e.getMessage()));
+    }
+  }
+
+  private static final String IMAGE_JPEG = "image/jpeg";
+
+  private static final String IMAGE_JPEG2K = "image/jp2";
+
+  private static final int THUMBNAIL_WIDTH = 200;
+
+  private static final int THUMBNAIL_HEIGHT = 200;
+
+  private static final long MEGABYTE = 1024L * 1024L;
+
+  private static final String JPG = "jpg";
+
+  private static final String JP2 = "jp2";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NitfPostProcessPlugin.class);
+
+  private static final String OVERVIEW = "overview";
+
+  private static final String ORIGINAL = "original";
+
+  private static final String DERIVED_IMAGE_FILENAME_PATTERN = "%s-%s.%s";
+
+  // non-word characters equivalent to [^a-zA-Z0-9_]
+  private static final String INVALID_FILENAME_CHARACTER_REGEX = "[\\W]";
+
+  private static final int ARGB_COMPONENT_COUNT = 4;
+
+  private static final int DEFAULT_MAX_NITF_SIZE = 120;
+
+  private static final int KB = 1024;
+
+  private static final int DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD = 32 * KB;
+
+  private boolean createOverview = true;
+
+  private boolean storeOriginalImage = true;
+
+  private volatile int maxNitfSizeMB = DEFAULT_MAX_NITF_SIZE;
+
+  private volatile double maxSideLength = DEFAULT_MAX_SIDE_LENGTH;
```

Comment:
❓ Why are `maxNitfSizeMB` and `maxSideLength` `volatile` but not other configurable properties?